core/parentchain/block-importer/src/parentchain_block_importer.rs

```diff
@@ -0,0 +1,234 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+//! Imports parentchain blocks and executes any indirect calls found in the extrinsics.
+
+use crate::{
+	beefy_merkle_tree::{merkle_root, Keccak256},
+	error::Result,
+};
+use ita_stf::ParentchainHeader;
```

Comment:
It seems odd to me that this is defined in the ita_std.

Reply:
I kind of agree, this was part of @gaudenzkessler 's PR that got merged recently

Reply:
The reason is probably that it's an alias around `sgx-runtime` types, which are defined in `ita-stf` already

Reply:
Yes - that's the result of a primitives mess. The sgx-runtime defines it's own types, even though they should be the same as the parentchain value. But this mess should be solved first on the sgx-runtime side before the worker can follow suit: https://github.com/integritee-network/sgx-runtime/issues/37